testing/python/issue/test_tilelang_issue_2942.py

```
@@ -0,0 +1,232 @@
+    @tilelang.jit
+    def prog():
+        @T.prim_func
+        def main():
+            with T.Kernel(1, threads=128):
+                f = T.alloc_fragment((M, N), "float32")
+                T.annotate_layout({f: invalid_layout})
+                T.clear(f)
+
+        return main
+
+    with pytest.raises(Exception, match="not injective|must map injectively"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)